inc/trilogy/protocol.h

```diff
@@ -412,25 +412,27 @@ typedef enum {
  * This should be sent in response to the initial handshake packet the server
  * sends upon connection.
  *
- * builder     - A pointer to a pre-initialized trilogy_builder_t.
- * user        - The username to use for authentication. Must be a C-string.
- * pass        - The password to use for authentication. Optional, and can be NULL.
- * pass_len    - The length of password in bytes.
- * auth_plugin - Plugin authentication mechanism that the server requested.
- * scramble    - The scramble value the server sent in the initial handshake.
- * flags       - Bitmask of TRILOGY_CAPABILITIES_t flags.
- *               The TRILOGY_CAPABILITIES_PROTOCOL_41 and
- *               TRILOGY_CAPABILITIES_SECURE_CONNECTION flags will always be set
- *               internally.
+ * builder          - A pointer to a pre-initialized trilogy_builder_t.
+ * user             - The username to use for authentication. Must be a C-string.
+ * pass             - The password to use for authentication. Optional, and can be NULL.
+ * pass_len         - The length of password in bytes.
+ * database         - The initial database to connect to. Optional, and can be NULL.
+ * client_encoding  - The charset to use for the connection. See TRILOGY_CHARSET_t.
+ * auth_plugin      - Plugin authentication mechanism that the server requested.
+ * scramble         - The scramble value the server sent in the initial handshake.
+ * flags            - Bitmask of TRILOGY_CAPABILITIES_t flags.
+ *                    The TRILOGY_CAPABILITIES_PROTOCOL_41 and
+ *                    TRILOGY_CAPABILITIES_SECURE_CONNECTION flags will always be set
+ *                    internally.
  *
  * Return values:
  *   TRILOGY_OK     - The packet was successfully built and written to the
  *                    builder's internal buffer.
  *   TRILOGY_SYSERR - A system error occurred, check errno.
  */
 int trilogy_build_auth_packet(trilogy_builder_t *builder, const char *user, const char *pass, size_t pass_len,
-                              const char *database, const char *auth_plugin, const char *scramble,
-                              TRILOGY_CAPABILITIES_t flags);
+                              const char *database, uint8_t client_encoding, const char *auth_plugin,
```

Comment:
This may not matter that much as I'm pretty sure it's a `uint8_t` under the hood, but for clarity should we use `TRILOGY_CHARSET_t` as the data type to be passed in the C API instead?